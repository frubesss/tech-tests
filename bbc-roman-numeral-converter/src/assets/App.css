.App {
  text-align: center;
  padding: 20px;
}

input.numeric-input-field {
  width: 150px;
}
