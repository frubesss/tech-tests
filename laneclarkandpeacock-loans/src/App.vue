<template>
    <div id="app">
        <headerNavigation/>
        <div class="app-content">
            <customerForm v-if="showCustomerForm"/>
            <agreementSummary v-else/>
        </div>
    </div>
</template>

<script>
    import headerNavigation from "./components/headerNavigation";
    import customerForm from "./components/customerForm"
    import agreementSummary from "./components/agreementSummary"

    export default {
        name: "App",
        components: {
            headerNavigation,
            customerForm,
            agreementSummary
        },
        computed: {
            showCustomerForm() {
                return this.$store.state.showCustomerForm
            }
        }
    };
</script>

<style>
    #app {
        font-family: 'Open Sans', sans-serif;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        color: #2c3e50;
    }

    .app-content {
        margin: 20px;
    }

</style>
