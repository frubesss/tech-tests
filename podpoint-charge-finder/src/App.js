import React, {Component} from 'react';
import Header from './components/header'
import Footer from './components/footer'
import Favourites from './components/favourites'
import Tabs from './components/tabs'

export default class App extends Component {

    constructor(props) {
        super(props);

        this.state = {
            favourites: [],
            buttonActive: 'FAVOURITES',
            isLoading: false
        };
        this.buttonClick = this.buttonClick.bind(this)
    }

    componentDidMount() {
        this.setState({isLoading: true});
        fetch('https://api.openchargemap.io/v2/poi/?client=ocm.app.web.5.2.7_20151008&verbose=false&output=json&operator=3&latitude=51.52351420000001&longitude=-0.0917564&distance=10&distanceunit=KM&maxresults=10')
            .then(response => response.json())
            .then(data => this.setState({favourites: data, isLoading: false}))
    }

    buttonClick(buttonName) {
        this.setState({
            buttonActive: buttonName
        });
    }

    render() {

        let {buttonActive, favourites, isLoading} = this.state;

        return (
            <div>
                <Header buttonActive={buttonActive}/>
                {buttonActive === 'FAVOURITES' &&
                <div>
                    <Tabs buttonClick={this.buttonClick} buttonActive={buttonActive}/>
                    <Favourites favourites={favourites} isLoading={isLoading}/>
                </div>
                }
                <Footer buttonActive={buttonActive} buttonClick={this.buttonClick}/>
            </div>
        );
    }
}