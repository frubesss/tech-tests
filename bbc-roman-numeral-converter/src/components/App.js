import React, {Component} from "react";
import RomanNumeralConverter from "./RomanNumeralConverter";
import "../assets/App.css";

export default class App extends Component {

    constructor(props) {
        super(props);
        this.state = {numberInserted: 0};
        this.onChange = this.onChange.bind(this);
    }

    onChange(event) {
        this.setState({numberInserted: parseInt(event.target.value, 10)});
    }

    render() {
        return (
            <div className="App">
                <input
                    className="numeric-input-field"
                    type="number"
                    min={0}
                    max={3999}
                    onChange={this.onChange}
                />
                <RomanNumeralConverter numberInserted={this.state.numberInserted}/>
            </div>
        );
    }
}
