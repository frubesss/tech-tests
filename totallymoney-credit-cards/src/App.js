import React from 'react';
import CreditCards from './components/creditCards';
import './assets/customerForm.scss'
import {CustomerForm} from './components/customerForm'

export default class App extends React.Component {

    state = {
        formSubmitted: false
    };

    submitForm = (values) => {
        this.setState({formSubmitted: true, customer: values})
    };

    render() {

        const {formSubmitted, customer} = this.state;

        return (
            <div>
            <img class="cat" src="https://www.catster.com/wp-content/uploads/2017/08/A-fluffy-cat-looking-funny-surprised-or-concerned.jpg"/>
                {formSubmitted ? <CreditCards customer={customer}/> : <CustomerForm submitForm={this.submitForm}/>}
            </div>
        );
    }
}