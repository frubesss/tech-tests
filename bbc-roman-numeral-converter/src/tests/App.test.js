import React from "react";
import {shallow} from "enzyme";
import App from "../components/App";
import renderer from "react-test-renderer";

describe('App Component', () => {

    let appComponent, input;

    beforeEach(() => {
        appComponent = shallow(<App/>);
    });

    it('Snapshot Test', () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Input Field', () => {

        beforeEach(() => {
            input = appComponent.find('input');
        });

        it('numberInserted state updated when numeric value inserted into input field', () => {
            input.simulate('change', {target: {value: 2}});
            expect(appComponent.state().numberInserted).toEqual(2);
        });

        it('has min value of 0 property', () => {
            expect(input.props().min).toEqual(0);
        });

        it('has max value of 3999 property', () => {
            expect(input.props().max).toEqual(3999);
        });

        it('has number type property', () => {
            expect(input.props().type).toEqual('number');
        });

    });

});
