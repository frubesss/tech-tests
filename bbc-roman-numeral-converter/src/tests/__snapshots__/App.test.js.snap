// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot Test 1`] = `
<div
  className="App"
>
  <input
    className="numeric-input-field"
    max={3999}
    min={0}
    onChange={[Function]}
    type="number"
  />
  <div>
    <h1>
      Please enter a number between 1 and 3999
    </h1>
  </div>
</div>
`;
